include/nbl/video/utilities/CReduce.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_VIDEO_C_REDUCE_H_INCLUDED_
+#define _NBL_VIDEO_C_REDUCE_H_INCLUDED_
+
+#include "nbl/video/utilities/CArithmeticOps.h"
+
+namespace nbl::video
+{
+
+    class CReduce final : public CArithmeticOps
+    {
+
+    public:
+        CReduce(core::smart_refctd_ptr<ILogicalDevice>&& device) : CReduce(std::move(device), core::roundDownToPoT(device->getPhysicalDevice()->getLimits().maxOptimallyResidentWorkgroupInvocations)) {}
```

Comment:
don't repeat the `roundDownToPoT` everywhere, you should rely on `CArithmeticOps` to set that up for you